src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/MatchingResults.java

```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol;
+
+import lombok.Data;
+
+@Data
+public class MatchingResults {
+    boolean validDate;
+    boolean validCaseId;
+    boolean validJurisdiction;
+    boolean validRegion;
+    boolean validLocation;
+    boolean validClassification;
```

Comment:
Consider that different types of Role Assignements can have different lists of attributes. For example `case id` does not apply for an organisational type role assignment, only for case role type role assignement.
Wondering whether modelling this as a boolean we wont' be able to distinguish whether the attribute was present but didn't match, or was not present at all
Maybe it won't matter. Let's give it a tought

Reply:
also, I'd rename to `caseIdMatched`, `jurisdictionMatched`.... since we are not doing a validation here

Reply:
update: confirmed that this won't work. The list of attributes is dynamic, so we need to either model them as Optional, or as a Map

Reply:
update: confirmed that this won't work. The list of attributes is dynamic, so we are going to model them as Optional, or as a Map. 
Based on those changes, do we also have to apply a similar change to this class?